recipes/cppwinrt/all/conanfile.py

```diff
@@ -0,0 +1,109 @@
+
+
+class CppWinRtConan(ConanFile):
+    name = "cppwinrt"
+    description = "C++17 language projection for Windows Runtime"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/microsoft/cppwinrt"
+    topics = ("native", "cpp", "winrt")
+    settings = "os", "arch", "compiler", "build_type"
+
+    def layout(self):
+        vs_layout(self)
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def requirements(self):
+        self.requires("winmd/1.0.210629.2")
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    def validate(self):
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        check_min_vs(self, 192)
+        if not is_msvc(self):
+            raise ConanInvalidConfiguration(
+                f"{self.ref} can be built only by Visual Studio and msvc.")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = MSBuildToolchain(self)
+        tc.properties["WindowsTargetPlatformVersion"] = "10.0"
+        tc.properties["WindowsTargetPlatformMinVersion"] = "10.0.17763.0"
+        tc.generate()
+        tc = MSBuildDeps(self)
+        tc.generate()
+        tc = VCVars(self)
+        tc.generate()
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+
+    @property
+    def msbuild_type(self):
+        return "Debug" if self.info.settings.build_type == "Debug" else "Release"
+
+    @property
+    def msbuild_folder(self):
+        return os.path.join(
+            self.build_folder, "_build", msbuild.msbuild_arch(self.info.settings.arch), self.msbuild_type)
+
+    def build(self):
+        self._patch_sources()
+        msbuild = MSBuild(self)
+        msbuild.build_type = self.msbuild_type
+        # use Win32 instead of the default value when building x86
+        msbuild.platform = "Win32" if self.info.settings.arch == "x86" else msbuild.platform
+        self.run(" ".join([msbuild.command(sln="cppwinrt.sln", targets=[
+                 "prebuild", "cppwinrt"]), f"/p:CppWinRTBuildVersion={self.version}"]))
+        self.run(" ".join([os.path.join(self.msbuild_folder, "cppwinrt.exe"),
+                    "-in", "local", "-out", self.msbuild_folder, "-verbose"]))
+
+    def package(self):
+        copy(self, pattern="LICENSE", dst=os.path.join(
+            self.package_folder, "licenses"), src=self.source_folder)
+        copy(
+            self, pattern="cppwinrt.exe", dst=os.path.join(self.package_folder, "bin"), src=self.msbuild_folder, keep_path=False
+        )
+        copy(
+            self,
+            pattern="*.h",
+            dst=os.path.join(self.package_folder, "include"),
+            src=os.path.join(self.msbuild_folder),
+            excludes="strings.h"
+        )
+
+    def package_info(self):
```

Comment:
Copied from the competitor: https://github.com/microsoft/vcpkg/blob/master/ports/cppwinrt/usage

Reply:
copied from the competitor: https://github.com/microsoft/vcpkg/blob/master/ports/cppwinrt/usage

Reply:
It's not official from the project https://github.com/microsoft/vcpkg/blob/master/ports/cppwinrt/portfile.cmake#L66 as far as I can tell.

In this case we recommend the default Conan targets, it's easier to remove them

This will have the same effect as what the other guys are doing, just the built-in way 😉

Reply:
How will have the same effect? Wouldn't the default target be `cppwinrt::cppwinrt`?

Reply:
The same effect as to provide unofficial CMake targets are a way of connivence 😄

Reply:
I think we can remove the targets and let the Conan client provide them

We generally only add them here as a reference to the "official" ones to make reviewing easier, this would lead to some confusion and it generally not recommended out side of ConanCenter